AI BLOCK and Documentation on How to add widget in MB

abhijeetsingh0401 opened the pull request:
-Added AI Block which connect the LLM to MB and Converts any abc song from LLM to MB
-Added Documentation on how to add a Widget to MB

abhijeetsingh0401 commented:
@walterbender @pikurasa could you review this branch?

pikurasa commented:
Right now, the AI widget is modeled after the sampler widget, using the sample block like this:

![Screenshot from 2024-08-12 14-19-02](https://github.com/user-attachments/assets/4c366515-1ef4-4053-9265-d9c49281b435)
 
Since the input and output for this widget has nothing to do with samples, I recommend doing something different. Perhaps it uses a print block instead, like this:

![Screenshot from 2024-08-12 14-22-15](https://github.com/user-attachments/assets/95484a2d-5224-47df-b0de-05c0d8cd30aa)

Notice that the print block's text block argument has the same text as the prompt.

@walterbender any input on this?

abhijeetsingh0401 commented:
@pikurasa @walterbender i have fixed the suggested implmentation please check also i have changed the localhost api to groq api when key added works great!

walterbender commented:
Looks good. @pikurasa let me know if there are more changes you need.

pikurasa commented:
> Looks good. @pikurasa let me know if there are more changes you need.

It will take me a few days before I can test this, but I think we can merge it if you think the code looks good and the latest changes pass the "smoke test".

pikurasa commented:
This is @abhijeetsingh0401 's final report: https://github.com/sugarlabs/musicblocks/pull/3964

Aditya-Raj-25 commented:
plzz assign me sir

pikurasa commented:
> plzz assign me sir

This is a PR, not an issue...

ZetaSpark commented:
Hi Sir, I am new to Sugar Labs and wants to contribute, Pls can you suggest where to start or give any "good first issue" if you have any?
